ha/cli/cortxha.py

```diff
@@ -17,106 +17,20 @@
 
 import os
 import sys
-import traceback
 import argparse
 import pathlib
 
-
-from cortx.utils.log import Log
-
-
-class Output:
-    def __init__(self):
-        """
-        Provide output for cluster command
-        """
-        self._output = "Success"
-        self._rc = 0
-
-    def output(self, output):
-        self._output = output
-
-    def rc(self, rc):
-        self._rc = rc
-
-    def get_output(self):
-        return self._output
-
-    def get_rc(self):
-        return self._rc
-
-#TODO - Move the cli to cortxcli framework once cortxcli is created as a separate module.
-class HACli:
+def main(argv):
     """
-    HACLI
+    Entry point for cortx CLI
     """
-    def __init__(self):
-        """
-        Initialization of HA CLI.
-        """
-        # TODO Check product env and load specific conf
-        ConfigManager.init('cortxha')
-
-        # get version from ha.conf
-        self._version = ConfigManager.get_major_version()
 
-    @staticmethod
-    def _usage():
-
-        if ha_cli._version == const.CORTX_VERSION_1:
-            return """
-    Use below command for more detail
-    cortxha --help
-            """
-        else:
-            return """
-    Use below command for more detail
-    cortx --help
-            """
-
-    def command(self):
-        try:
-            argParser = argparse.ArgumentParser(
-                usage = "%(prog)s\n\n" +  HACli._usage(),
-                formatter_class = argparse.RawDescriptionHelpFormatter)
-
-            component_parser = argParser.add_subparsers(
-                help = "Select one of given component.",
-                dest = "cortxha_action")
+    sys.path.append(os.path.join(os.path.dirname(pathlib.Path(__file__)), '..', '..'))
 
-            CommandFactory.get_command(component_parser)
-            args = argParser.parse_args()
+    from ha.cli.commands import Command
+    command = Command()
+    command.process(argv[1:])
```

Comment:
I suspect this code will fail with default exception when no arguments are passed.

Reply:
This failure case  is handled in Command.parse()  
So this returns the exception  HAInvalidCommand  if no arguments are passed

Reply:
Agree. My fault - the slice`[1:]` will be empty list if no arguments are passed.